packages/server/src/adapters/node-http/nodeHTTPRequestHandler.ts

```diff
@@ -123,14 +79,24 @@ export async function nodeHTTPRequestHandler<
         });
       },
       contentTypeHandler,
-      unstable_onHead,
-      unstable_onChunk,
     });
 
-    if (isStream) {
-      opts.res.write(formatter!.end());
-      opts.res.end();
+    // res is of type `Request` but node.js wants us to call res.send etc
+
+    if (!opts.res.statusCode || opts.res.statusCode === 200) {
+      opts.res.statusCode = res.status;
+    }
+    opts.res.statusMessage = res.statusText;
+    for (const [key, value] of res.headers.entries()) {
+      opts.res.setHeader(key, value);
+    }
+
+    if (res.body) {
```

Comment:
A check of `Symbol.asyncIterator in res.body`, would at least assert that the body is an `AsyncIterable`.